wherobots/db/cursor.py

```diff
@@ -43,8 +43,32 @@ def __get_results(self) -> Optional[list[Any]]:
         result = self.__queue.get()
         if isinstance(result, DatabaseError):
             raise result
+        type_map = {
+            'object': 'STRING',
+            'int64': 'NUMBER',
+            'float64': 'NUMBER',
+            'datetime64[ns]': 'DATETIME',
+            'timedelta[ns]': 'DATETIME',
+            'bool': 'NUMBER',  # Assuming boolean is stored as number
+            'bytes': 'BINARY',
+        }
         self.__rowcount = len(result)
         self.__results = result
+        if not result.empty:
+            self.__description = [
+                (
+                    col_name, # name
+                    type_map.get(str(result[col_name].dtype), 'STRING'),  # type_code
+                    None,  # display_size
+                    result[col_name].memory_usage(),  # internal_size
+                    None,  # precision
+                    None,  # scale
+                    True  # null_ok; Assuming all columns can accept NULL values
+                )
+                for col_name in result.columns
+            ]
+        else:
+            self.__description = None
```

Comment:
Since you already (re)set `self.__description` to `None` in the `execute()` method, I don't think you need this here. `None` will be the default value, and a description will only get set when it's needed (when there are non-empty results).